app/main.py

```diff
@@ -1 +1,45 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __new__(cls, *args, **kwargs):
+        cls.alive.append(obj := object.__new__(cls))
+        return obj
+
+    def __init__(self, name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    @property
+    def health(self) -> int:
+        return self.__health
+
+    @health.setter
+    def health(self, health: int) -> None:
+        if health > 0:
+            self.__health = health
+        else:
+            self.__health = 0
+            self.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, other: Animal) -> None:
```

Comment:
I think you need use @staticmethod, you don't use `self` in this function.

Reply:
I don't think so: static method may be called directly from the class.
`Carnivore.bite()`, however, is intended to be used only from class instances.

Reply:
You still can call this function with class instances when this function have @staticmethod, and with this method you don't have unuse `self`. I think it's best practise.

Reply:
Now you can call its function with class instances(for example `lion.bite(rabit)`) but can't with class `Carnivore.bite(rabit)`.
With @staticmethod you can call with class instances and with class.

Reply:
> You still can call this function with class instances when this function have @staticmethod

Yes, I know. But the reason I think the method shouldn't be static is that I don't want it to be used directly from the class. E.g.:
```python
Carnivore.bite(rabbit)  # TypeError: Carnivore.bite() missing 1 required positional argument: 'other'
```

Reply:
Ok, maybe you right)  Sorry for that remark